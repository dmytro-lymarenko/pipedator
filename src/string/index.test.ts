import { string } from './index';

describe('string()', () => {
	it('should validate a string', () => {
		expect(string().validate('')).toBeNull();
	});

	it('should return an error for non-string values', () => {
		expect(string().validate(5)).toMatchSnapshot();
		expect(string().validate({})).toMatchSnapshot();
		expect(string().validate([])).toMatchSnapshot();
		expect(string().validate(() => {})).toMatchSnapshot();
		expect(string().validate(null)).toMatchSnapshot();
		expect(string().validate(undefined)).toMatchSnapshot();
	});

	it('should support custom message', () => {
		expect(string('Custom message').validate(5)).toMatchSnapshot();
	});
});
