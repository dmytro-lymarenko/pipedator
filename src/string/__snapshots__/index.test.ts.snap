// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`string() should return an error for non-string values 1`] = `
Object {
  "children": null,
  "message": "Value should be a string",
  "path": "",
}
`;

exports[`string() should return an error for non-string values 2`] = `
Object {
  "children": null,
  "message": "Value should be a string",
  "path": "",
}
`;

exports[`string() should return an error for non-string values 3`] = `
Object {
  "children": null,
  "message": "Value should be a string",
  "path": "",
}
`;

exports[`string() should return an error for non-string values 4`] = `
Object {
  "children": null,
  "message": "Value should be a string",
  "path": "",
}
`;

exports[`string() should return an error for non-string values 5`] = `
Object {
  "children": null,
  "message": "Value should be a string",
  "path": "",
}
`;

exports[`string() should return an error for non-string values 6`] = `
Object {
  "children": null,
  "message": "Value should be a string",
  "path": "",
}
`;

exports[`string() should support custom message 1`] = `
Object {
  "children": null,
  "message": "Custom message",
  "path": "",
}
`;
